README.md

```diff
@@ -171,6 +171,50 @@ That said, even if your application runs on Lambda, any core instrumentation iss
+
+Also generally, bundlers work by crawling all of the `require()` calls that an application makes to files on disk, replacing the `require()` calls with custom code, and then concatenating all of the resulting JavaScript into one "bundled" file. When a built-in module is loaded, like `require('fs')`, that call can then remain the same in the resulting bundle.
+
+Fundamentally APM tools like `dd-trace` stop working at this point. Perhaps they continue to intercept the calls for built-in modules but don't intercept calls to third party libraries. This means that by default when you bundle a `dd-trace` app with a bundler it is likely to capture information about disk access (via `fs`) and outbound HTTP requests (via `http`), but will otherwise omit calls to third party libraries (like extracting incoming request route information for the `express` framework or showing which query is run for the `mysql` database client).
+
+To get around this, one can treat all third party modules, or at least third party modules that the APM needs to instrument, as being "external" to the bundler. With this setting the instrumented modules remain on disk and continue to be loaded via `require()` while the non-instrumented modules are bundled. Sadly this results in a build with many extraneous files and starts to defeat the purpose of bundling. 
+
+For these reasons it's necessary to have custom-built bundler plugins. Such plugins are able to instruct the bundler on how to behave, injecting intermediary code and otherwise intercepting the "translated" `require()` calls. The result is that many more packages are then included in the bundled JavaScript file. Some applications can have 100% of modules bundled, however native modules still need to remain external to the bundle.
+
+### Esbuild Support
+
+This library provides experimental esbuild support in the v3 release line and requires at least Node.js v14.17. To use the bundler, make sure you have `dd-trace` installed, and then require the `dd-trace/esbuild` module when building your bundle.
+
+Here's an example of how one might use `dd-trace` with esbuild:
+
+```javascript
+const ddPlugin = require('dd-trace/esbuild')
+const esbuild = require('esbuild')
+
+esbuild.build({
+  entryPoints: ['app.js'], 
+  bundle: true,
+  outfile: 'out.js',
+  plugins: [ddPlugin],
+  external: [ // this depends on the native modules used by your app
+    'pg-native',
+    'graphql/language/visitor',
+    'graphql/language/printer',
+    'graphql/utilities',
+  ],
+  platform: 'node', // allows built-in modules to be required
+  target: ['node16']
+}).catch((err) => {
+  console.error(err)
+  process.exit(1)
+})
+```
+
+When you run your build, if you get errors when encountering native modules, you'll need to add them to the `external` list. In the above example the application is using the native Postgres library as well as some native GraphQL libraries.
```

Comment:
Is this paragraph still needed if all of `node_modules` is externalized?

Reply:
Wasn't the plan to only suggest externalizing the things to be instrumented? Externalizing _everything_ kind of defeats to purpose of bundling. 🤷🏻

Reply:
True, I guess externalizing everything is actually an alternative to using the bundler plugin.

@tlhunter Wouldn't native addons always be externalized? Is esbuild able to bundle native code somehow?

Reply:
You're right Roch, the paragraph wouldn't be needed if all of `node_modules` were externalized. You're right Stephen, we certainly don't want to externalize everything. You're right Roch, externalizing everything could be considered an alternative to the plugin.

Roch, it looks like the `esbuild-node-externals` package you mentioned earlier would automatically externalize all of the encountered native modules. I could simply recommend that and then be done with it.

Bryan, you're probably right, certainly there's nothing special about dd-trace when it comes to native modules as compared to any other tool. I'm not familiar what is common knowledge in the community or what is idiomatic documentation for esbuild libraries.

Taking a step back, any users that want to take their existing bundled app and introduce dd-trace will _already_ be doing something to get around the native modules. Also, any users wishing to switch their app over to being bundled will _already_ need to research how to handle native modules. For that reason I'll simply delete the paragraph outright.